pom.xml

```diff
@@ -230,6 +230,22 @@ THE SOFTWARE.
         <artifactId>access-modifier-annotation</artifactId>
         <version>${access-modifier-annotation.version}</version>
       </dependency>
+      <dependency>
+        <!--  make sure these old servlet versions are never used by us or by any plugins which end up depening on this version -->
+        <!-- Jenkins test Harness tries to fudge this, causes issues with some IDEs -->
```

Comment:
Is some matching PR to `jenkins-test-harness` needed here then?

Reply:
JTH already uses servlet 3.x so that should be OK.

The plugin POM should have the hack for servlet 2.x vs 3.x removed but that will mean it will no longer work correctly with old cores (ones where we originally added 3.1 support) as plugin authors may end up with servlet 3.1 on the classpath for compilation but end up using deploying to a jenkins that uses 2.4

I'm ok with creating a new plugin-pom and removing support for any jenkins in the pre 3.1 servlet era.  (basically pre 2.0 (2016-04-20) )